sympy/assumptions/tests/test_query.py

```diff
@@ -2027,18 +2026,19 @@ def test_key_extensibility():
     raises(AttributeError, lambda: ask(Q.my_key(x)))
 
     # Old handler system
-    class MyAskHandler(AskHandler):
-        @staticmethod
-        def Symbol(expr, assumptions):
-            return True
-    try:
-        register_handler('my_key', MyAskHandler)
-        assert ask(Q.my_key(x)) is True
-        assert ask(Q.my_key(x + 1)) is None
-    finally:
-        remove_handler('my_key', MyAskHandler)
-        del Q.my_key
-    raises(AttributeError, lambda: ask(Q.my_key(x)))
+    with warns_deprecated_sympy():
```

Comment:
This context manager should be used specifically around the line that emits the warning.

It only tests that the warning is emitted at least once so if there are multiple lines that warn then it needs to be used multiple times like:
```
with warns():
    line1
with warns():
    line2
```